src/parser/jobs/pld/modules/Goring.tsx

```diff
@@ -1,62 +1,300 @@
+import {t} from '@lingui/macro'
 import {Trans} from '@lingui/react'
-import {ActionLink, StatusLink} from 'components/ui/DbLink'
-import ACTIONS from 'data/ACTIONS'
-import STATUSES from 'data/STATUSES'
+import {DataLink, ActionLink} from 'components/ui/DbLink'
+import {Event, Events} from 'event'
+import {Analyser} from 'parser/core/Analyser'
+import {filter} from 'parser/core/filter'
 import {dependency} from 'parser/core/Injectable'
-import Checklist, {Requirement, Rule} from 'parser/core/modules/Checklist'
-import {DoTs} from 'parser/core/modules/DoTs'
-import Suggestions, {SEVERITY, TieredSuggestion} from 'parser/core/modules/Suggestions'
-import React from 'react'
-
-const SEVERITIES = {
-	CLIPPING: {
-		2500: SEVERITY.MINOR,
-		5000: SEVERITY.MEDIUM,
-		10000: SEVERITY.MAJOR,
-	},
+import {Actors} from 'parser/core/modules/Actors'
+import Checklist, {Rule, Requirement} from 'parser/core/modules/Checklist'
+import {Data} from 'parser/core/modules/Data'
+import {Invulnerability} from 'parser/core/modules/Invulnerability'
+import {Statuses} from 'parser/core/modules/Statuses'
+import Suggestions from 'parser/core/modules/Suggestions'
+import React, {ReactNode} from 'react'
+import {Accordion, Table} from 'semantic-ui-react'
+
+const MAX_ALLOWED_CLIPPING = 3000
+
+interface DotApplicationData {
+	event: Events['statusApply'],
+	clip?: number,
+	source: number
 }
 
-export class Goring extends DoTs {
-	static override handle = 'goring'
+interface DotStatusData {
+	lastApplication: number,
+	applications: DotApplicationData []
+}
+
+interface DotTargetData {
+	[key: number]: DotStatusData,
```

Comment:
```suggestion
	[statusId: Status['id']]: DotStatusData,
```

Reply:
More to the point, don't write records like this.

```ts
// Prefer (record type)
type DotTargetData = Record<Status['id'], DotStatusData>
// Over (index signature)
interface DotTargetData {
	[statusId: Status['id']]: DotStatusData
}
```

I'd generally recommend against using index types in basically all scenarios.